### PR TITLE
Demo breakup 2

### PR DESCRIPTION
**Pull request by nkohen**
The second leg of #1140 (built on top of #1146)

Adds `--collateral` flag to `createdlcoffer`

Introduces `initdlcmutualclose`

Added `fromPath(path: BIP32Path)` methods to HD things (`HDAccount`, `HDCoin`, etc)

Created DLCSetupMessage trait to make getting common info from offer and accept messages possible

Fixed remote collateral setting

Fixed bugs

Created DLCWallet method to create `BinaryOutcomeDLCClient` with `SetupDLC` from db data

### CODE REVIEW

File: core/src/main/scala/org/bitcoins/core/hd/HDAccount.scala [hunk 1]
```diff
@@ -28,6 +28,18 @@ case class HDAccount(
 
 object HDAccount {
 
+  def fromPath(path: BIP32Path): Option[HDAccount] = {
+
+    HDCoin.fromPath(BIP32Path(path.path.dropRight(1))).flatMap { coin =>
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/hd/HDAccount.scala]: nit: `path.path.dropRight(1)` -> `path.path.init`

File: dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala [hunk 2]
```diff
@@ -1077,8 +1089,8 @@ object BinaryOutcomeDLCClient {
       remoteInput = offer.totalCollateral,
       fundingUtxos = fundingUtxos,
       remoteFundingInputs = offer.fundingInputs,
-      winPayout = offer.contractInfo.values.head,
-      losePayout = offer.contractInfo.values.last,
+      winPayout = offer.contractInfo.values.last,
+      losePayout = offer.contractInfo.values.head,
```
Comment (benthecarman) [inline dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala:1093]: This doesn't seem right, we are doing the following for the corresponding hashes

```
outcomeWin = offer.contractInfo.keys.head,
outcomeLose = offer.contractInfo.keys.last,
```

File: core/src/main/scala/org/bitcoins/core/hd/HDCoinType.scala [hunk 3]
```diff
@@ -33,4 +33,11 @@ object HDCoinType {
       case _: Int =>
         throw new IllegalArgumentException(s"$int is not valid coin type!")
     }
+
+  def fromNode(node: BIP32Node): HDCoinType = {
+    require(node.hardened,
+            s"Cannot construct HDPurpose from un-hardened node: $node")
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/hd/HDCoinType.scala]: > HDPurpose

HDCoinType

File: dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala (hunk 2 above)
Comment (nkohen) [reply dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala:1093]: Right but `contractInfo` contains the payouts of the counter-party

**Review by benthecarman — APPROVED** [review]

**Review by Christewart — APPROVED** [review]

**Pull request merged**